autotest/utilities/test_gdalalg_raster_color_merge.py

```diff
@@ -0,0 +1,362 @@
+    grayscale_ds = gdal.GetDriverByName("MEM").Create("", 1, 1)
+    with pytest.raises(
+        Exception, match="Only 3 or 4-band Byte dataset supported as input"
+    ):
+        gdal.Run(
+            "raster",
+            "color-merge",
+            input=input_ds,
+            grayscale=grayscale_ds,
+            output_format="stream",
+        )
+
+    input_ds = gdal.GetDriverByName("MEM").Create("", 1, 1, 3)
+    grayscale_ds = gdal.GetDriverByName("MEM").Create("", 1, 1, 2)
+    with pytest.raises(
+        Exception, match="Only 1-band Byte dataset supported as grayscale dataset"
+    ):
+        gdal.Run(
+            "raster",
+            "color-merge",
+            input=input_ds,
+            grayscale=grayscale_ds,
+            output_format="stream",
+        )
+
+    input_ds = gdal.GetDriverByName("MEM").Create("", 1, 1, 3)
+    grayscale_ds = gdal.GetDriverByName("MEM").Create("", 1, 1, 1, gdal.GDT_UInt16)
+    with pytest.raises(
+        Exception, match="Only 1-band Byte dataset supported as grayscale dataset"
+    ):
+        gdal.Run(
+            "raster",
+            "color-merge",
+            input=input_ds,
+            grayscale=grayscale_ds,
+            output_format="stream",
+        )
+
+    input_ds = gdal.GetDriverByName("MEM").Create("", 2, 10, 3)
+    grayscale_ds = gdal.GetDriverByName("MEM").Create("", 2, 9, 1)
+    with pytest.raises(
+        Exception,
+        match="Input RGB/RGBA dataset and grayscale dataset must have the same dimensions",
+    ):
+        gdal.Run(
+            "raster",
+            "color-merge",
+            input=input_ds,
+            grayscale=grayscale_ds,
+            output_format="stream",
+        )
+
+    input_ds = gdal.GetDriverByName("MEM").Create("", 9, 2, 3)
+    grayscale_ds = gdal.GetDriverByName("MEM").Create("", 10, 2, 1)
+    with pytest.raises(
+        Exception,
+        match="Input RGB/RGBA dataset and grayscale dataset must have the same dimensions",
+    ):
+        gdal.Run(
+            "raster",
+            "color-merge",
+            input=input_ds,
+            grayscale=grayscale_ds,
+            output_format="stream",
+        )
+
+
+@pytest.mark.parametrize(
+    "xsize,ysize",
+    [
+        (None, None),
+        ((1 << 31) / 2, 1),
+        ((1 << 31) - 1, (1 << 31) - 1),
+        (100 * 1000 * 1000, 100 * 1000 * 1000),
+    ],
+)
+@pytest.mark.parametrize("req_object", ["dataset", "band"])
+def test_gdalalg_raster_color_merge_out_of_memory(tmp_vsimem, xsize, ysize, req_object):
+
+    if xsize == 100 * 1000 * 1000 and gdaltest.is_travis_branch("sanitize"):
+        # otherwise it would throw AddressSanitizer: out of memory: allocator is trying to allocate 0x8e1bc9bf040000 bytes
+        pytest.skip("skip on ASAN")
+
+    gdal.GetDriverByName("GTiff").Create(
+        tmp_vsimem / "in.tif", 2, 2, 3, options=["COMPRESS=DEFLATE", "BLOCKYSIZE=2"]
+    )
+    with gdal.VSIFile(tmp_vsimem / "in.tif", "rb") as f:
```

Comment:
why you read data from a newly created file and rewrite them to the same file?

Reply:
I actually truncate the file by writing all bytes but the last 3 ones, to make it invalid and check correct I/O error propagation. I'll add a code comment